Source/Tests/LinuxSyscalls/Syscalls.cpp

```diff
@@ -226,89 +211,156 @@ static bool IsSupportedByInterpreter(std::string const &Filename) {
   return false;
 }
 
-uint64_t ExecveHandler(const char *pathname, char* const* argv, char* const* envp, ExecveAtArgs *Args) {
+static bool IsShebangFilename(std::string const &Filename) {
+  // Open the Filename to determine if it is a shebang file.
+  std::fstream File;
+  size_t FileSize{0};
+  File.open(Filename, std::fstream::in | std::fstream::binary);
+
+  if (!File.is_open())
+    return false;
+
+  File.seekg(0, File.end);
+  FileSize = File.tellg();
+  File.seekg(0, File.beg);
+
+  // Is the file large enough for shebang
+  if (FileSize <= 2)
+    return false;
+
+  // The maximum length of the shebang line is `#!` + 255 chars
+  std::vector<char> Header(std::min(257ul, FileSize));
+  File.read(&Header.at(0), Header.size());
```

Comment:
```suggestion
  std::array<char, 257> Header;
  auto ChunkSize = std::min(257ul, FileSize);
  File.read(Header.data(), ChunkSize);
```

Reply:
Some of the logic with this comment and others conflicted. So you'll need to check that again with the changes.